agent/fw_wordpress.c

```diff
@@ -578,6 +580,65 @@ NR_PHP_WRAPPER(nr_wordpress_apply_filters) {
 }
 NR_PHP_WRAPPER_END
 
+#if ZEND_MODULE_API_NO >= ZEND_7_4_X_API_NO
+/*
+ * Wrap the wordpress function add_filter
+ *
+ * function add_filter( $hook_name, $callback, $priority = 10, $accepted_args = 1 )
+ *
+ * @param string   $hook_name     The name of the filter to add the callback to.
+* @param callable $callback      The callback to be run when the filter is applied.
+* @param int      $priority      Optional. Used to specify the order in which the functions
+*                                associated with a particular filter are executed.
+*                                Lower numbers correspond with earlier execution,
+*                                and functions with the same priority are executed
+*                                in the order in which they were added to the filter. Default 10.
+* @param int      $accepted_args Optional. The number of arguments the function accepts. Default 1.
+* @return true Always returns true.
+ */
+
+NR_PHP_WRAPPER(nr_wordpress_add_filter) {
+  /* Wordpress's add_action() is just a wrapper around add_filter(),
+   * so we only need to instrument this function */
+  NR_UNUSED_SPECIALFN;
+  (void)wraprec;
+  const char* filename = NULL;
+  bool wrap_hook = true;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_WORDPRESS);
+
+  /*
+   * We only want to hook the function being called if this is a WordPress
+   * function, we're instrumenting hooks, and WordPress is currently executing
+   * hooks (denoted by the wordpress_tag being set).
+   */
+  if ((NR_FW_WORDPRESS != NRPRG(current_framework))
+      || (0 == NRINI(wordpress_hooks))) {
+    wrap_hook = false;
+  }
+
+  if (NRINI(wordpress_hooks_skip_filename)
+      && 0 != nr_strlen(NRINI(wordpress_hooks_skip_filename))) {
+    filename = nr_php_op_array_file_name(NR_OP_ARRAY);
+
+    if (nr_strstr(filename, NRINI(wordpress_hooks_skip_filename))) {
+      nrl_verbosedebug(NRL_FRAMEWORK, "skipping hooks for function from %s",
+                       filename);
+      wrap_hook = false;
+    }
+  }
```

Comment:
This whole block of code is exactly the same as [this block](https://github.com/newrelic/newrelic-php-agent/blob/41bb3cd04d28d4472130717207cee71033bef064/agent/fw_wordpress.c#L386-L405) in `nr_wordpress_call_user_func_array`. The only difference is in how filename is retrieved. I think it makes sense to extract this into a helper function, say `static inline bool nr_wordpress_should_wrap_hook_callback(const char* filename)` to avoid this code duplication.

Then the function would start:
```
   char* filename = nr_php_op_array_file_name(<op_array_source>);

   if (nr_wordpress_should_wrap_hooks_callback(filename)) {
       <func-specific-wrapping>
   }
```

Note that the call to `nr_php_op_array_file_name` is 'cheap' because it is just pointer assignment, not `strcpy` or `strdup`, and therefore does not need to be wrapped with `if (is-wordpress-ini-set)`.

Reply:
This will not be done to limit number of conflicts when syncing oapi branch.